app/src/main/java/org/dhis2/usescases/eventsWithoutRegistration/eventCapture/EventCaptureFragment/EventCaptureFormPresenter.kt

```diff
@@ -60,12 +62,27 @@ class EventCaptureFormPresenter(
                 .subscribeOn(schedulerProvider.io())
                 .observeOn(schedulerProvider.ui())
                 .subscribe(
-                    { view.showFields(it, lastFocusItem) },
+                    { fields ->
+                        view.showFields(fields, lastFocusItem)
+                        selectedSection ?: fields
+                            .mapNotNull { it.programStageSection() }
+                            .firstOrNull()
+                            ?.let{ selectedSection = it }
+                    },
```

Comment:
Event capture form presenter has already unit test for this. I think it would be very simple and worthy to try to add some checks to this new behaviour